clients/src/test/java/org/apache/kafka/common/record/FileRecordsTest.java

```diff
@@ -279,7 +279,7 @@ public void testSearch() throws IOException {
      * Test that the message set iterator obeys start and end slicing
      */
     @Test
-    public void testIteratorWithLimits() throws IOException {
+    public void testIteratorWithLimits() {
```

Comment:
Some of the `FileRecords` are not closed during the test; should we also clean them?
L368, L382

Reply:
We have 100s of unclosed resources in tests. I'm not saying we shouldn't try to fix them, but maybe we can do that in other PRs.